### PR TITLE
Allow to deactivate package managers

### PR DESCRIPTION
**Pull request by sschuberth**
Please have a look at the individual commit messages for the details.

### CODE REVIEW

File: cli/src/main/kotlin/commands/AnalyzerCommand.kt [hunk 1]
```diff
@@ -142,14 +142,24 @@ class AnalyzerCommand : CliktCommand(name = "analyze", help = "Determine depende
...
-        help = "The comma-separated package managers to activate, any of ${allPackageManagersByName.keys}."
+        help = "The comma-separated package managers to activate, any of ${allPackageManagersByName.keys}. Note that " +
+                "deactivation overrides activation."
     ).convert { name ->
         allPackageManagersByName[name]
             ?: throw BadParameterValue("Package managers must be one or more of ${allPackageManagersByName.keys}.")
     }.split(",").default(PackageManager.ALL)
 
+    private val deactivatedPackageManagers by option(
+        "--no-package-managers", "-n",
```
Comment (tsteenbe) [inline cli/src/main/kotlin/commands/AnalyzerCommand.kt:155]: `--no-package-managers` sound weird to me, using `--disable-package-managers` or `--deactivate-package-managers` make more sense to me.

**Review by tsteenbe — DISMISSED** [review]

File: cli/src/main/kotlin/commands/AnalyzerCommand.kt (hunk 1 above)
Comment (sschuberth) [reply cli/src/main/kotlin/commands/AnalyzerCommand.kt:155]: I chose a long option name that starts with "n" as I liked the short option name to be `-n`, which sounds like a "not"-operator, which actually makes some sense. And I liked that "n" alphabetically comes after "m" to resemble that the `-m` and `-n` options are related here 😀

Comment (tsteenbe) [reply cli/src/main/kotlin/commands/AnalyzerCommand.kt:155]: Understood but is it then better to change to `--not-package-managers`?

Comment (sschuberth) [reply cli/src/main/kotlin/commands/AnalyzerCommand.kt:155]: I'd be fine with that, will change.

**Review by oheger-bosch — APPROVED** [review]

**Pull request merged**